docs/content/docs/dev/table/sql/queries/hints.md

```diff
@@ -84,4 +84,210 @@ insert into kafka_table1 /*+ OPTIONS('sink.partitioner'='round-robin') */ select
 
 ```
 
+## Query Hints
+
+`Query Hints` specify that the indicated hints are used in the scope of a query. Their effective scope is current `Query 
```

Comment:
I haven't find a formal definition of 'Query Block' in the standard, but some descriptions from vendors:
[ref1](https://sqlmaria.com/2020/03/10/what-are-query-block-names-and-how-to-find-them/)
[ref2](https://www.ibm.com/docs/en/ida/9.1.1?topic=explain-query-blocks)

the query block example in ref1 looks very clear to me, shall we add an example to illustrate the query block scope(ref2 was more formal but not easy to understand)? it may helps users to get the concept quickly, WDYT?

cc @godfreyhe any thoughts on this?